### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by kaisong1990**
When creating a new PR against Common, there will be 2 new pipelines running to check if this PR will break MSAL C++ for macOS and iOS. 

Currently those two pipelines will checkout out the latest dev branch from MSAL C++ repo  and use their template yml files to run build and test on both platforms.

Remaining works is to let MSAL C++ branch use the commit from the PR as the subtree and do renaming. I have created a separate [ticket](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1130164) to work on the feature

### CODE REVIEW

**Comment by kaisong1990** [comment]
Close and reopen to do an e2e test

File: azure_pipelins/verify_msalcpp_per_pr_mac.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  # - task: UsePythonVersion@0
+  #   displayName: 'Use Python 3.x'
+  # - task: PythonScript@0
+  #   displayName: Ensure generated Podspec for macOS is up to date
+  #   inputs:
+  #     scriptPath: scripts/update_msalobjc_subtree.py
+
+  - template: azure_pipelines/templates/macos-setup.yml@microsoft-authentication-library-for-cpp
+
+  - template: azure_pipelines/templates/build-and-test.yml@microsoft-authentication-library-for-cpp
```
Comment (ruixi1994) [inline azure_pipelins/verify_msalcpp_per_pr_mac.yml:53]: Looks we are testing ios but not macOS, any particular reason for that? I think we can do build only for both platform so far. And we could have follow up about what tests are necessary to be run. My concern is many of those tests are testing shared code which could add running time for this CI and not necessary with common core pr submit. We can probably split some of them out to run there. Thanks

File: azure_pipelins/verify_msalcpp_per_pr_mac.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+variables:
+  runCodesignValidationInjection: false
+
+jobs:
+- job: macOS_OnPrCommit
+  displayName: x64 Debug
+  pool:
+    vmImage: macOS-10.15
+  timeoutInMinutes: 40
```
Comment (ruixi1994) [inline azure_pipelins/verify_msalcpp_per_pr_mac.yml:31]: If we move out tests, this timeout could be decreased to 20 minutes

**Review by ruixi1994 — APPROVED** [review]

**Comment by ruixi1994** [comment]
I think we can disable it now for common repo pr submit since it is currently not running on the latest commits yet. And let's enable after we got the subtree updates and renaming work done in part of the CI.

File: azure_pipelins/verify_msalcpp_per_pr_mac.yml (hunk 1 above)
Comment (kaisong1990) [reply azure_pipelins/verify_msalcpp_per_pr_mac.yml:53]: As discussed, I have removed the build and publish from mac

File: azure_pipelins/verify_msalcpp_per_pr_ios.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+resources:
+ repositories:
+   - repository: microsoft-authentication-library-for-cpp
+     type: github
+     endpoint: 'GitHub for AzureAD and Azure-Samples (as aadidgit service)'
+     name: AzureAD/microsoft-authentication-library-for-cpp
+     ref: develop
+
+variables:
+  runCodesignValidationInjection: false
```
Comment (NerevarineRule) [inline azure_pipelins/verify_msalcpp_per_pr_ios.yml:24]: Is this to address the known issue that we have with CI/CD not recognizing MacOS profile? current workaround is to simply change it on XCODE to "sign in to run locally"

File: azure_pipelins/verify_msalcpp_per_pr_ios.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  #   inputs:
+  #     scriptPath: scripts/update_msalobjc_subtree.py
+
+  - template: azure_pipelines/templates/ios-setup.yml@microsoft-authentication-library-for-cpp
+
+  - template: azure_pipelines/templates/build-and-test.yml@microsoft-authentication-library-for-cpp
+    parameters:
+      configuration: Debug
+      platform: iOS
+      arch: x64
```
Comment (NerevarineRule) [inline azure_pipelins/verify_msalcpp_per_pr_ios.yml:56]: Just curious - why do we not need to also provide arch support for arm?

**Review by NerevarineRule — APPROVED** [review]

**Comment by NerevarineRule** [comment]
LGTM. Take it easy ;)

**Comment by oldalton** [comment]
@kaisong1990, Can you provide a little bit more context in the PR description about how it works? e.g. does it use CPP dev branch etc?

**Comment by kaisong1990** [comment]
> @kaisong1990, Can you provide a little bit more context in the PR description about how it works? e.g. does it use CPP dev branch etc?

Updated above

File: azure_pipelins/verify_msalcpp_per_pr_ios.yml (hunk 3 above)
Comment (kaisong1990) [reply azure_pipelins/verify_msalcpp_per_pr_ios.yml:24]: Hi Peter, sorry for the late replying, this tag is not related to iOS specific

File: azure_pipelins/verify_msalcpp_per_pr_ios.yml (hunk 4 above)
Comment (kaisong1990) [reply azure_pipelins/verify_msalcpp_per_pr_ios.yml:66]: Could you clarify the question? What is arm here?

**Pull request merged**